Fix system test scaffold generator for vowels

msducheminjr opened the pull request:
Fix issue where running "rails g scaffold Author" will create system
test names like "creating a Author" and "updating a Author" so that
there are not grammatical errors. Now running "rails g scaffold Author"
will generate "should create Author", "should update Author", and
"should destroy Author" which works around the article vowel-sound
agreement issues.

Fixes #40744

### Summary

<!-- Provide a general description of the code changes in your pull
request... were there any bugs you had fixed? If so, mention them. If
these bugs have open GitHub issues, be sure to tag them here as well,
to keep the conversation linked together. -->

### Other Information
Also needed to update the testing guide.
<!-- If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

If you are updating any of the CHANGELOG files or are asked to update the
CHANGELOG files by reviewers, please add the CHANGELOG entry at the top of the file.

Finally, if your pull request affects documentation or any non-code
changes, guidelines for those changes are [available
here](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#contributing-to-the-rails-documentation)

Thanks for contributing to Rails! -->

rafaelfranca commented:
@dhh do you have any opinion about this? This change makes sense to me.

jonathanhefner commented:
Instead of "my", what about "a new" / "an existing"?

msducheminjr commented:
@jonathanhefner that would also work. I initially went with something that would keep the generated test name as concise as possible. I like the indefinite article (inflected properly) better than "my", but in the spirit of agile, wanted to solve with a small incremental improvement.

Another option would be to word it like the controller tests and use ```"should create ..."```, ```"should update"```, etc.

eileencodes commented:
Functional tests use `should create` so I'd lean towards that

```
railties/lib/rails/generators/test_unit/scaffold/templates/functional_test.rb.tt
13:  test "should get index" do
18:  test "should get new" do
23:  test "should create <%= singular_table_name %>" do
31:  test "should show <%= singular_table_name %>" do
36:  test "should get edit" do
41:  test "should update <%= singular_table_name %>" do
46:  test "should destroy <%= singular_table_name %>" do
```

msducheminjr commented:
If everybody is good with the ```"should create/update/destroy <%= human_name %>"```, I can update. 

Anybody opposed to that approach?

msducheminjr committed:
Fix system test scaffold generator for vowels

Fix issue where running "rails g scaffold Author" will create system
test names like "creating a Author" and "updating a Author" so that
there are not grammatical errors. Now running "rails g scaffold Author"
will generate "should create Author", "should update Author", and
"should destroy Author" which works around the article vowel-sound
agreement issues.

Fixes #40744
